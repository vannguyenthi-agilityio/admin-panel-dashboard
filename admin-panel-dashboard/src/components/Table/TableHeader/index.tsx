import { memo } from "react";

// Components
import HeaderCell from "../HeaderCell";

// Types
import { IColumnType, ICustomerTable } from "@/types";

interface IPropsDataTableHeader {
  columns: IColumnType<ICustomerTable>[];
  onSort: (key: string) => void;
}

const TableHeader = ({
  columns,
  onSort,
}: IPropsDataTableHeader) => (
  <thead>
    <tr>
      {columns.map(({ key, title, isSortable = false }) => {
        const handleClickSort = () => {
          isSortable && onSort(key);
        };

        return (
          <th
            data-testid={`header-cell-${title}`}
            onClick={handleClickSort}
            key={key}
            className={`px-6 py-2 leading-[17px] tracking-[0.2px] ${
              isSortable ? "cursor-pointer" : "cursor-default"
            }`}>
            <HeaderCell
              title={title}
              isSortable={isSortable}
            />
          </th>
        );
      })}
    </tr>
  </thead>
);

export default memo(TableHeader);
