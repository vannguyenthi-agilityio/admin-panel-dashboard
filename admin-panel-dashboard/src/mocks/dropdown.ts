// Types
import { IListDropDown, IUser } from "@/types";

export const MOCK_USER_DATA: IUser = {
  name: "Derek Alvarado",
  avatar: "https://www.gravatar.com/avatar/2c7d99fe281ecd3bcd65ab915bac6dd5?s=250"
}

export const LIST_DROPDOWN_DATA: IListDropDown[] = [
  {
    href: "/",
    name: "Dashboard"
  },
  {
    href: "/",
    name: "Settings"
  }
]
