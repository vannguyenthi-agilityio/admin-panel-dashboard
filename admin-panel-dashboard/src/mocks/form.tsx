// Types
import { ICustomerData } from "@/types";

export const MOCK_CUSTOMER_DATA: ICustomerData = {
  id: 1,
  idNumber: 123,
  email: "test@gmail.com",
  firstName: "Alyvia",
  lastName: "Kelley",
  phoneNumber: "20 1234 4567",
  dateOfBirth: {months: '', date: 1, years: 1}
}

export const MOCK_INIT_CUSTOMER_DATA: ICustomerData = {
  id: "",
  idNumber: 0,
  email: "",
  firstName: "",
  lastName: "",
  phoneNumber: "",
  dateOfBirth: {months: '', date: 0, years: 0}
}
