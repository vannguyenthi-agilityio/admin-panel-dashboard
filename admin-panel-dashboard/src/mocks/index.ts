export * from './dropdown';
export * from './table';
export * from './form';
