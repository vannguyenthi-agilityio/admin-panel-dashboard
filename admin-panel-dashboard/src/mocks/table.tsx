// Types
import { IColumnType, ICustomerTable } from "@/types";

export const MOCK_CUSTOMER: ICustomerTable[] = [
  {
    id: 1,
    dateOfBirth: "06/18/1978",
    status: 'Approved',
    email: "test@gmail.com",
    fullName: "Alyvia Kelley",
  },
  {
    id: 2,
    dateOfBirth: "09/18/1983",
    status: 'Rejected',
    email: "jaiden.n@gmail.com",
    fullName: "Jaiden Nixon",
  },
  {
    id: 3,
    dateOfBirth: "10/18/1988",
    status: 'Blocked',
    email: "me@clayton.com",
    fullName: "Jaiden Nixon",
  }
];

export const MOCK_COLUMNS: IColumnType<ICustomerTable>[] = [
  {
    key: "id",
    title: "#",
  },
  {
    key: "fullName",
    title: "Full Name",
    isSortable: true,
  },
  {
    key: "status",
    title: "Status",
    isSortable: false,
  },
  {
    key: "email",
    title: "E-Mail",
    isSortable: true,
  },
  {
    key: "dateOfBirth",
    title: "Date of Birth",
    isSortable: true,
  },
];
