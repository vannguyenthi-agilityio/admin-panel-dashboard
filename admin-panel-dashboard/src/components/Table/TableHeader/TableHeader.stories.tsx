// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import TableHeader from ".";

// Mocks
import { MOCK_COLUMNS } from "@/mocks";

// Types
import { IColumnType } from "@/types";

const meta: Meta<typeof TableHeader> = {
  title: "Components/Table/TableHeader",
  tags: ["autodocs"],
  component: TableHeader,
  argTypes: {
    onSort: { description: "funct sort of table" },
    columns: { description: "columns of header table" },
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    columns: MOCK_COLUMNS as IColumnType<unknown>[],
    onSort: () => {},
  },
};
