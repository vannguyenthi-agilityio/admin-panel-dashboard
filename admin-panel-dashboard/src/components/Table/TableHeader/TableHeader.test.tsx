import { render } from "@testing-library/react";

// Mocks
import { MOCK_COLUMNS } from "@/mocks";

// Components
import TableHeader from ".";

const mockFunction = jest.fn();

describe("TableHeader", () => {
  const mockProps = {
    columns: MOCK_COLUMNS,
    onSort: mockFunction
  };

  it("Should matches snapshot", async () => {
    const { container } = render(
      <table>
        <TableHeader {...mockProps} />
      </table>,
    );
    expect(container).toMatchSnapshot();
  });
});
