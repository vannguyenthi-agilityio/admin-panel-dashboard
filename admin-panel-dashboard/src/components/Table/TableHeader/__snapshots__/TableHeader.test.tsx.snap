// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader Should matches snapshot 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-default"
          data-testid="header-cell-#"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            >
              #
            </div>
          </div>
        </th>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-pointer"
          data-testid="header-cell-Full Name"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            >
              Full Name
            </div>
            <button
              aria-label="Arrow Sort"
              class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-transprent border-none hover:bg-transprent text-sm rounded-lg w-[16px] h-[16px] p-0 ml-2"
              type="button"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.66669 5.16666H13.3334M4.33335 7.83332H11.6667M5.66669 10.5H10.3334"
                  stroke="#6c757d"
                  stroke-linecap="round"
                />
              </svg>
            </button>
          </div>
        </th>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-default"
          data-testid="header-cell-Status"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            >
              Status
            </div>
          </div>
        </th>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-pointer"
          data-testid="header-cell-E-Mail"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            >
              E-Mail
            </div>
            <button
              aria-label="Arrow Sort"
              class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-transprent border-none hover:bg-transprent text-sm rounded-lg w-[16px] h-[16px] p-0 ml-2"
              type="button"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.66669 5.16666H13.3334M4.33335 7.83332H11.6667M5.66669 10.5H10.3334"
                  stroke="#6c757d"
                  stroke-linecap="round"
                />
              </svg>
            </button>
          </div>
        </th>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-pointer"
          data-testid="header-cell-Date of Birth"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            >
              Date of Birth
            </div>
            <button
              aria-label="Arrow Sort"
              class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-transprent border-none hover:bg-transprent text-sm rounded-lg w-[16px] h-[16px] p-0 ml-2"
              type="button"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.66669 5.16666H13.3334M4.33335 7.83332H11.6667M5.66669 10.5H10.3334"
                  stroke="#6c757d"
                  stroke-linecap="round"
                />
              </svg>
            </button>
          </div>
        </th>
        <th
          class="px-6 py-2 leading-[17px] tracking-[0.2px] cursor-default"
          data-testid="header-cell-"
        >
          <div
            class="flex items-center w-full"
          >
            <div
              class="text-sm font-normal text-darker"
            />
          </div>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
